server/src/org/apache/cloudstack/network/lb/CertServiceImpl.java

```diff
@@ -339,190 +348,182 @@ public SslCertResponse createCertResponse(SslCertVO cert, List<LoadBalancerCertM
         return response;
     }
 
-    private void validateCert(Certificate cert, boolean chainPresent) {
+    private void validateCert(final Certificate cert, final boolean chainPresent) {
 
-        if (!(cert instanceof X509Certificate))
+        if (!(cert instanceof X509Certificate)) {
             throw new IllegalArgumentException("Invalid certificate format. Expected X509 certificate");
+        }
 
         try {
             ((X509Certificate)cert).checkValidity();
-        } catch (Exception e) {
+        } catch (final Exception e) {
             throw new IllegalArgumentException("Certificate expired or not valid", e);
         }
     }
 
-    private void validateKeys(PublicKey pubKey, PrivateKey privKey) {
+    private void validateKeys(final PublicKey pubKey, final PrivateKey privKey) {
 
-        if (pubKey.getAlgorithm() != privKey.getAlgorithm())
+        if (pubKey.getAlgorithm() != privKey.getAlgorithm()) {
             throw new IllegalArgumentException("Public and private key have different algorithms");
+        }
 
         // No encryption for DSA
-        if (pubKey.getAlgorithm() != "RSA")
+        if (pubKey.getAlgorithm() != "RSA") {
             return;
+        }
 
         try {
 
-            String data = "ENCRYPT_DATA";
-            SecureRandom random = new SecureRandom();
-            Cipher cipher = Cipher.getInstance(pubKey.getAlgorithm());
+            final String data = "ENCRYPT_DATA";
+            final SecureRandom random = new SecureRandom();
```

Comment:
Re-initializing a `SecureRandom` instance on every method call is expensive and unnecessary.  Please consider refactoring to a `static` constant and reusing the singleton.